src/transformers/models/blip/modeling_blip.py

```diff
@@ -1191,10 +1204,17 @@ def forward(
 
         question_embeds = question_embeds[0] if not return_dict else question_embeds.last_hidden_state
 
-        if decoder_input_ids is None:
-            decoder_input_ids = torch.LongTensor([self.decoder_bos_token_id]).repeat((batch_size, 1))
+        if labels is not None and decoder_input_ids is None:
+            # get decoder inputs from shifting lm labels to the right - this is used in training mode
+            decoder_input_ids = self._shift_right(labels)
+            # replace possible -100 values in labels by `pad_token_id`
+            labels = labels.masked_fill(labels == self.decoder_pad_token_id, -100)
+        elif decoder_input_ids is None:
+            # by default use BOS token as decoder_input_ids
+            decoder_input_ids = torch.LongTensor([self.decoder_start_token_id]).repeat((batch_size, 1))
 
         if labels is None:
+            # labels is None, but decoder_input_ids is not None, this is used for inference
```

Comment:
I'm also not sure there's a need for these lines, see BART and T5 who don't have this either

Reply:
No model in Transformers creates `labels` when the user doesn't specify them ;)

Reply:
THanks! Fixed in 61309e9